microcosm/config/model.py

```diff
@@ -98,26 +98,44 @@ def __init__(
         self.default_factory = default_factory
         self.mock_value = mock_value
         self.required = required
-        self.nullable = nullable
-
-        # Warn when the user doesn't provide exactly one of
-        #
-        # [`required=True`, `default_value=...`]
-        #
-        # When the user is not using `default_factory`, this could be legacy
-        # code, so we just warn. In the future this will be an error, but for
-        # now we just warn in those situations which could occur as of the time
-        # this was introduced
+        self.nullable = nullable or (default_value is None)
```

Comment:
`default_value is None` now implies `nullable=True` so that existing examples like the following work:

```py
foo=required(int, default_value=None)
```

Reply:
Yeah, removing redundancy is nicer.